scripts/parse-coverage.sh

```
@@ -0,0 +1,175 @@
+while IFS= read -r line; do
+  # Skip mode line
+  [[ "$line" =~ ^mode: ]] && continue
+  [[ -z "$line" ]] && continue
+
+  # Extract file path and data
+  file=$(echo "$line" | awk '{print $1}')
+  data=$(echo "$line" | awk '{print $2, $3, $4, $5}')
+
+  # Extract package path: strip module prefix and filename
+  # github.com/poyrazk/thecloud/internal/core/services/instance.go -> internal/core/services
+  pkg="${file#$MODULE_PREFIX}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)